admin/tests-integration/azure-table.data.service.spec.js

```diff
@@ -0,0 +1,161 @@
+'use strict'
+
+/* global describe expect beforeAll afterAll fail it */
+
+const { TableClient, TableServiceClient } = require('@azure/data-tables')
+const config = require('../config')
+const connectionString = config.AZURE_STORAGE_CONNECTION_STRING
+const uuid = require('uuid')
+const sut = require('../services/data-access/azure-table.data.service')
+const RA = require('ramda-adjunct')
+
+const commonPrefix = 'mtcIntegrationTest'
+const testRunTableNames = []
+
+function getUniqueTableName () {
+  let id = uuid.v4()
+  id = id.replace(/-/g, '')
+  const name = `${commonPrefix}${id}`
+  testRunTableNames.push(name)
+  return name
+}
+
+async function createTable (tableName) {
+  const client = TableClient.fromConnectionString(connectionString, tableName)
+  return client.createTable()
+}
+
+async function deleteTable (tableName) {
+  const client = TableClient.fromConnectionString(connectionString, tableName)
+  return client.deleteTable()
+}
+
+describe('azure-table.data.service', () => {
+  beforeAll(async () => {
+    const client = TableServiceClient.fromConnectionString(connectionString)
+    const iterator = client.listTables()
+    const integrationTestTables = []
+    for await (const table of iterator) {
+      if (table.name.startsWith(commonPrefix)) {
+        integrationTestTables.push(table.name)
+      }
+    }
+    const deletions = integrationTestTables.map(t => {
+      return client.deleteTable(t)
+    })
+    await Promise.all(deletions)
+  })
+  afterAll(async () => {
+    try {
+      const deletions = testRunTableNames.map(t => {
+        return deleteTable(t)
+      })
+      await Promise.all(deletions)
+    } catch (error) {
+      fail(`failed to delete one or more tables, azure storage may need manual clean up.\n${error}`)
+    }
+  })
+
+  describe('retrieveEntity', () => {
+    it('returns raw entity when exists', async () => {
```

Comment:
really new tests should be written in jest, so `test` rather than `it` is preferred.